src/settings/LocationLocations/LocationForm/ServicePointsFields.js

```diff
@@ -95,52 +81,54 @@ class ServicePointsFields extends React.Component {
         </Layout>
         <Layout className={`display-flex ${css.radioButtonLayout}`}>
           <Field
-            component={this.radioButtonComp}
+            component={radioButtonComp}
             fieldIndex={index}
             name={`${field}.primary`}
           />
         </Layout>
       </Layout>
     );
-  }
-
-  render() {
-    const { formValues } = this.props;
+  };
 
-    // make the last existing service point to be the primary one
-    if (formValues.servicePointIds && formValues.servicePointIds.length === 1 && !formValues.servicePointIds[0].primary) {
-      this.singlePrimary(0);
-    }
+  // Make the last existing service point to be the primary one
+  if (formValues.servicePointIds && formValues.servicePointIds.length === 1 && !formValues.servicePointIds[0].primary) {
```

Comment:
Suggestion: From a readability perspective, it could be useful to move it into a separate constant or even functions checker.